core/src/main/java/com/coderanch/blackjack/Hand.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.blackjack;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.function.Predicate.not;
+
+import static com.coderanch.blackjack.Card.Rank.ACE;
+import static com.coderanch.util.require.Require.requireThat;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * A hand in a game of Blackjack.
+ */
+final class Hand {
+
+    /**
+     * Highest score in Blackjack.
+     */
+    static final int MAX_LEGAL_SCORE = 21;
+
+    /**
+     * The cards in the hand.
+     */
+    private final List<Card> cards;
+
+    /**
+     * Creates a new hand.
+     *
+     * @param firstCard  first card dealt.
+     * @param secondCard second card dealt
+     * @throws IllegalArgumentException if either {@code firstCard} or {@code secondCard} is {@code null}.
+     */
+    Hand(Card firstCard, Card secondCard) {
+        this(List.of(
+            requireThat("firstCard", firstCard, is(notNullValue())),
+            requireThat("secondCard", secondCard, is(notNullValue()))
+        ));
+    }
+
+    /**
+     * Constructs a new hand with the given cards.
+     *
+     * @param cards the cards in the hand.
+     */
+    private Hand(List<Card> cards) {
+        this.cards = cards;
+    }
+
+    /**
+     * Gets a new hand with the extra card.
+     *
+     * @param card the card being added to the hand.
+     * @return a new hand with the extra card.
+     * @throws IllegalArgumentException if {@code card} is {@code null}.
+     */
+    Hand withAdditionalCard(Card card) {
+        requireThat("card", card, is(notNullValue()));
+
+        var newCards = new ArrayList<Card>(this.cards);
+        newCards.add(card);
+        return new Hand(newCards);
+    }
+
+    /**
+     * Gets the best score.
+     *
+     * @return the best score.
```

Comment:
Requires more explanation.

```suggestion
     * @return the highest score that can be made by using a large point value for aces
     *         without the hand going bust, or the lowest score if the hand is bust.
```